File: kattis_done/3dprinter/3dprinter.js
'use strict'
const readline = require('readline');


const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

rl.on('line', (line) => {
	var nums = line.split(' ');
	let n = parseInt(nums[0]);
	let d = [];
	for (let i = 0; i < n + 1; i++) {
		d.push([]);
		for (let j = 0; j < n + 1; j++) {
			d[i].push(-1);
		}
	}
	function recurse(printer, statue) {
		if (printer >= statue) return 1;
		if (d[printer][statue] == -1) {
			d[printer][statue] = Math.min(recurse(printer * 2, statue), recurse(printer, statue - printer))
		}
		return 1 + d[printer][statue];
	}
	console.log(recurse(1, n));
});

File: kattis/3dprinter/3dprinter.cpp
#include <bits/stdc++.h>

using namespace std;
vector<vector<int> > d;

int recurse(int printer, int statue) {
	if (printer >= statue) return 1;
	if (d[printer][statue] == -1) d[printer][statue] = min({recurse(printer * 2, statue), recurse(printer, statue - printer)});
	return 1 + d[printer][statue];
}
int main() {
	int n;
	cin >> n;
	d = vector<vector<int> > (n + 1, vector<int>(n + 1, -1));
	cout << recurse(1, n) << endl;
	return 0;
}